scripts/neighbor_advertiser

```diff
@@ -0,0 +1,307 @@
+    platform_summary = str(show_platform_summary.decode(DEFAULT_DECODING))
+
+    for line in platform_summary.split('\n'):
+        if line.strip().startswith('HwSKU:'):
+            hwsku = line.split(' ')[1]
+            break
+
+    return hwsku
+
+
+def get_vlan_interfaces():
+    vlan_interfaces = []
+
+    ls_all_interfaces = subprocess.check_output(['ls', NET_PATH])
+    all_interfaces = str(ls_all_interfaces.decode(DEFAULT_DECODING))
+
+    for interface in all_interfaces.split('\n'):
+        if interface.strip().startswith(VLAN_PATH_PREFIX):
+            vlan_interfaces.append(interface.strip())
+
+    return vlan_interfaces
+
+
+def get_vlan_interface_mac_address(vlan_if):
+    cat_mac_address = subprocess.check_output(['cat', NET_PATH + vlan_if + VLAN_ADDRESS_PATH])
+    mac_address = str(cat_mac_address.decode(DEFAULT_DECODING)).strip()
+
+    return mac_address
+
+
+def get_vlan_interface_vlan_id(vlan_if):
+    return vlan_if[4:]
+
+
+def get_vlan_interface_vxlan_id(vlan_if):
+    return vlan_if[4:]
+
+
+def get_neighbor_advertiser_addresses():
+    cfggen_na_addr = subprocess.check_output(['sonic-cfggen', '-d', '-t', TEMPLATE_PATH])
+    na_addr = str(cfggen_na_addr.decode(DEFAULT_DECODING))
+
+    return na_addr
+
+
+def get_lo_inet_addr(na_addr):
+    for line in na_addr.split('\n'):
+        if 'Loopback0 inet addr' in line:
+            lo_inet_addr = line.split(' ')[3]
+            break
+
+    return lo_inet_addr
+
+
+def get_lo_inet6_addr(na_addr):
+    for line in na_addr.split('\n'):
+        if 'Loopback0 inet6 addr' in line:
+            lo_inet6_addr = line.split(' ')[3]
+            break
+
+    return lo_inet6_addr
+
+
+def get_vlan_inet_addr(vlan_if, na_addr):
+    vlan_inet_addr = []
+
+    prefix = vlan_if + ' inet addr'
+    for line in na_addr.split('\n'):
+        if prefix in line:
+            vlan_inet_addr.append(line.split(' ')[3])
+            break
+
+    return vlan_inet_addr
+
+
+def get_vlan_inet6_addr(vlan_if, na_addr):
+    vlan_inet6_addr = []
+
+    prefix = vlan_if + ' inet6 addr'
+    for line in na_addr.split('\n'):
+        if prefix in line:
+            vlan_inet6_addr.append(line.split(' ')[3])
+            break
+
+    return vlan_inet6_addr
+
+
+def save_as_json(obj, file_path):
+    dir_path = os.path.dirname(file_path)
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+
+    with open(file_path, 'w') as outfile:
+        json.dump(obj, outfile, sort_keys = True)
+
+
+def construct_vxlan_tunnel(ferret_ip):
+    na_addr = get_neighbor_advertiser_addresses()
+    lo_inet_addr = get_lo_inet_addr(na_addr)
+    vxlan_tunnel_full = {}
+
+    vxlan_tunnel_base = {
+        VXLAN_TUNNEL_NAME: {
+            'src_ip': lo_inet_addr,
+            'dst_ip': ferret_ip
+        }
+    }
+
+    vxlan_tunnel_maps = {}
+    vlan_interfaces = get_vlan_interfaces()
+    index = 0
+
+    for vlan_interface in vlan_interfaces:
```

Comment:
you could use `enumerate` here to have an index of the element and traversed object element